packages/ckeditor5-ui/src/dropdown/utils.js

```diff
@@ -291,6 +291,97 @@ export function focusChildOnDropdownOpen( dropdownView, childSelectorCallback )
 	}, { priority: 'low' } );
 }
 
+export function addKeyboardHandlingForGrid( view, gridElementsCollection, numberOfColumns, isTable ) {
+	view.keystrokes.set( 'arrowright', ( evt ) => {
+		const gridElements = [ ...gridElementsCollection ];
+		const focusedElementIndex = getFocusedElement( gridElements );
+		let nextIndex;
+
+		// TODO: for tables'd be ok, for others should not change row no
+		if ( focusedElementIndex === gridElements.length - 1 ) {
+			nextIndex = 0;
+		} else {
+			nextIndex = focusedElementIndex + 1;
+		}
+
+		gridElements[ nextIndex ].focus();
+
+		if ( isTable ) {
+			gridElements[ nextIndex ].selectTile( view );
+		}
+
+		evt.stopPropagation();
+		evt.preventDefault();
+	} );
+
+	view.keystrokes.set( 'arrowleft', ( evt ) => {
+		const gridElements = [ ...gridElementsCollection ];
+		const focusedElementIndex = getFocusedElement( gridElements );
+		let nextIndex;
+
+		// TODO: for tables'd be ok, for others should not change row no
+		if ( focusedElementIndex === 0 ) {
+			nextIndex = gridElements.length - 1;
+		} else {
+			nextIndex = focusedElementIndex - 1;
+		}
+
+		gridElements[ nextIndex ].focus();
+
+		if ( isTable ) {
+			gridElements[ nextIndex ].selectTile( view );
+		}
+
+		evt.stopPropagation();
+		evt.preventDefault();
+	} );
+
+	view.keystrokes.set( 'arrowup', ( evt ) => {
+		const gridElements = [ ...gridElementsCollection ];
+		const focusedElementIndex = getFocusedElement( gridElements );
+		let nextIndex = focusedElementIndex - numberOfColumns;
+
+		if ( nextIndex < 0 ) {
+			nextIndex = focusedElementIndex + numberOfColumns * Math.floor( gridElements.length / numberOfColumns );
+			if ( nextIndex > gridElements.length -1 ) {
+				nextIndex -= numberOfColumns;
+			}
+		}
+
+		gridElements[ nextIndex ].focus();
+
+		if ( isTable ) {
+			gridElements[ nextIndex ].selectTile( view );
+		}
+
+		evt.stopPropagation();
+		evt.preventDefault();
+	} );
+
+	view.keystrokes.set( 'arrowdown', ( evt ) => {
+		const gridElements = [ ...gridElementsCollection ];
+		const focusedElementIndex = getFocusedElement( gridElements );
+		let nextIndex = focusedElementIndex + numberOfColumns;
+
+		if ( nextIndex > gridElements.length - 1 ) {
+			nextIndex = focusedElementIndex % numberOfColumns;
+		}
+
+		gridElements[ nextIndex ].focus();
+
+		if ( isTable ) {
+			gridElements[ nextIndex ].selectTile( view );
+		}
+
+		evt.stopPropagation();
+		evt.preventDefault();
+	} );
+
+	function getFocusedElement( gridElements ) {
```

Comment:
I chose the first option for two reasons:

1.  Less code if we have the focus tracker only in helper (not in every grid view)
2.  In case of new grid view we won't force the developer to remember that the focus tracker has to be created

Reply:
Eventually I chose the second option because some grids already have their own focusTrackers and we can use them.